provider/online.go

```diff
@@ -34,6 +33,19 @@ type OnlineStoreTable interface {
 	Get(entity string) (interface{}, error)
 }
 
+type VectorStore interface {
+	CreateIndex(feature, variant, entity string, vectorType VectorType) (VectorStoreTable, error)
+	GetIndex(feature, variant string) (string, error) // TODO: determine the type of index
+	DeleteIndex(feature, variant string) error
+	Close() error
+	Provider
+}
+
+type VectorStoreTable interface {
+	OnlineStoreTable
```

Comment:
This was genius, good thinking

Reply:
🤣 This was _your_ idea, but I'll take whatever I can get.